addon-resizer/nanny/kubernetes_client.go

```diff
@@ -53,6 +72,45 @@ func (k *kubernetesClient) CountNodes() (uint64, error) {
 	return uint64(len(result.Items)), err
 }
 
+func (k *kubernetesClient) countNodesThroughMetrics() (uint64, error) {
+	rawMetrics, err := k.clientset.Core().RESTClient().Get().RequestURI("/metrics").DoRaw()
+	if err != nil {
+		return 0, err
+	}
+
+	decoder := expfmt.SampleDecoder{
+		Dec:  expfmt.NewDecoder(strings.NewReader(string(rawMetrics)), expfmt.FmtText),
+		Opts: &expfmt.DecodeOptions{},
+	}
+	var v model.Vector
+	for {
+		if err := decoder.Decode(&v); err != nil {
+			if err == io.EOF {
+				break
+			}
+			continue
+		}
+		for _, metric := range v {
+			name := metric.Metric[model.MetricNameLabel]
+			if name != objectCountMetricName {
+				continue
+			}
+			resource := metric.Metric["resource"]
```

Comment:
NIT: Given that we have a constant for "nodes" we should also introduce one for "resource"

Reply:
done